client/question/components/LearnerViewDirective.js

```diff
@@ -925,6 +1037,69 @@ tie.directive('learnerView', [function() {
           $scope.scrollToTopOfFeedbackWindow();
         };
 
+        /**
+         * Sets the code in the code editor to the snapshot index
+         * passed in as a parameter.
+         *
+         * @param {number} selectedSnapshotIndex The snapshot index
+         * selected in the previous snapshots dropdown.
+         */
+        $scope.revertToSelectedSnapshot = function(selectedSnapshotIndex) {
+          $scope.currentSnapshotIndex = selectedSnapshotIndex;
+          var selectedSnapshot = null;
+          if (selectedSnapshotIndex === 0) {
+            selectedSnapshot = SessionHistoryService.getStarterCodeSnapshot();
+            $scope.previousButtonIsDisabled = true;
+          } else {
+            selectedSnapshot = SessionHistoryService.getPreviousSnapshot(
+              selectedSnapshotIndex);
+            $scope.previousButtonIsDisabled = false;
+          }
+          if (selectedSnapshot === null) {
+            throw Error('Could not retrieve code for snapshot at index ' +
+              selectedSnapshotIndex);
+          } else {
+            $scope.editorContents.code = selectedSnapshot;
+          }
+          $scope.snapshotMenuIsOpen = false;
+        };
+
+        /**
+         * Sets the code in the code editor to the previous snapshot.
+         */
+        $scope.revertToPreviousSnapshot = function() {
+          var previousSnapshotIndex = $scope.currentSnapshotIndex - 1;
```

Comment:
Is there any reason this method can't just be the previous method but with $scope.currentSnapshotIndex - 1?